### PR TITLE
Upgrade to Jupyter Server 2

### PR DESCRIPTION
**Pull request by shreve**
My strategy is to update the dependency spec then fix whatever tests start failing.

### CODE REVIEW

**Comment by luisarturopena** [comment]
Any updates on this? 😃

**Comment by brichet** [comment]
Thanks @shreve.

That would be a regression from https://github.com/jupyter/nbgrader/pull/1586, not sure this is what we want.
Maybe it will be the way to go for this branch (Jupyterlab 3.x and Notebook 6.x).

cc. @jtpio @blink1073 who may have an opinion on it.

File: nbgrader/server_extensions/assignment_list/handlers.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
 from textwrap import dedent
 
 from jupyter_server.utils import url_path_join as ujoin
-from jupyter_server.base.handlers import JupyterHandler
 from traitlets import Unicode, default
 from traitlets.config import LoggingConfigurable, Config
 from jupyter_core.paths import jupyter_config_path
+from notebook.base.handlers import IPythonHandler as JupyterHandler
```
Comment (jtpio) [inline nbgrader/server_extensions/assignment_list/handlers.py:15]: Why import the handler from the `notebook` and not `jupyter_server`?

**Comment by blink1073** [comment]
I'd say the notebook 6.x compat branch should keep its pin on `jupyter_server<2`, and then use `>=2.0` for notebook 7.x going forward.

**Comment by lahwaacz** [comment]
The main branch already has `jupyter_server>=2`: https://github.com/jupyter/nbgrader/commit/1b5ffa93db15012f15ac17bf475eab2420fe73bf#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711

**Pull request closed without merging**